subsys/net/ip/net_if.c

```diff
@@ -4186,6 +4186,9 @@ int net_if_up(struct net_if *iface)
 	if (status < 0) {
 		goto out;
 	}
+#if defined(CONFIG_NET_IPV4_IGMP)
+	net_ipv4_igmp_init(iface);
+#endif /* CONFIG_NET_IPV4_IGMP */
```

Comment:
Perhaps a separate function that checks also that IPv4 is enabled for this interface.
```
void init_igmp(struct net_if *iface)
{
#if defined(CONFIG_NET_IPV4_IGMP)
    if (iface->config.ip.ipv4 == NULL) {
     return;
   }

  net_ipv4_igmp_init(iface);
#else 
  return;
#endif 
}
```

Reply:
Thanks, I've incorporated this into my latest change.